### DISCUSSION TITLE
[Python][Arrow] Pushdown for offset/limit

### DISCUSSION THREAD

**Discussion by changhiskhan** [duckdb/duckdb#5066] [General]
Is there's a way to push down offset/limit to the data access layer?
We have a special pyarrow scanner subclass that knows how to read the exact byte range for offset/limit. So if there are no aggregates, it would be really efficient to push the offset/limit down as extra ScanOptions for arrow (esp if the projection columns are large blobs like image bytes or something).

**Comment by hannes** [comment]
No, this is currently not possible.

**Comment by arjenpdevries** [comment]
Similar in spirit to a "positional join", a physical operator that would also make some columnstore functionality really faster (I think), without requiring an index, by using order and exact sizes to know where to read. Are such operators on the road map Hannes?
